Audio buffer was stalled at first segment

### What version of Hls.js are you using?
1.0.0
### What browser and OS (including versions) are you using?
Windows Chrome /Mac OS Chrome/Windows Edge
### Test stream:
https://os2man2.github.io/sample_hls_fmp4/master.m3u8

https://hls-js.netlify.app/demo/?src=https%3A%2F%2Fos2man2.github.io%2Fsample_hls_fmp4%2Fmaster.m3u8&demoConfig=eyJlbmFibGVTdHJlYW1pbmciOnRydWUsImF1dG9SZWNvdmVyRXJyb3IiOnRydWUsInN0b3BPblN0YWxsIjpmYWxzZSwiZHVtcGZNUDQiOmZhbHNlLCJsZXZlbENhcHBpbmciOi0xLCJsaW1pdE1ldHJpY3MiOi0xfQ==

<!-- If possible, please provide a test stream or page -->
<!-- You can paste your stream into the demo and provide the permalink here -->

### Configuration:

<!-- Describe the hls.js configuration and any additional player setup steps -->

```
{
  "debug": true,
  "enableWorker": true,
  "lowLatencyMode": true,
  "backBufferLength": 90
}
```

### Checklist

<!-- Replace [ ] with [x] to check off the list -->

- [x] The issue observed is not already reported by searching on Github under https://github.com/video-dev/hls.js/issues
- [x] The issue occurs in the stable client on https://hls-js.netlify.com/demo and not just on my page
<!-- The stable client is built from the latest release -->
- [x] The issue occurs in the latest client on https://hls-js-dev.netlify.com/demo and not just on my page
<!-- The latest client is built from the head of the master branch -->
- [x] The stream has correct Access-Control-Allow-Origin headers (CORS)
- [x] There are no network errors such as 404s in the browser console when trying to play the stream

### Steps to reproduce

1. See playing demo stream
2. Problem was happen every times.

### Expected behavior
Play video without stop stream 

### Actual behavior
Stop stream and cannot continue to play it any more.
Because audio buffer was stalled at first segment 
when changing audio group from l0 to l3. 
### Console output

```
Using Hls.js config: Object
logger.ts:74 [log] >
hls.ts:354 [log] > stopLoad
hls.ts:330 [log] > loadSource:https://test-streams.mux.dev/x36xhzz/x36xhzz.m3u8
stream-controller.ts:540 [log] > [stream-controller]: Trigger BUFFER_RESET
hls.ts:302 [log] > attachMedia
buffer-controller.ts:736 [log] > [buffer-controller]: Media source opened
base-stream-controller.ts:1265 [log] > [subtitle-stream-controller]: STOPPED->IDLE
level-controller.ts:171 [log] > [level-controller]: manifest loaded, 5 level(s) found, first bitrate: 2149280
buffer-controller.ts:129 [log] > 1 bufferCodec event(s) expected
hls.ts:344 [log] > startLoad(-1)
level-controller.ts:250 [log] > [level-controller]: switching to level 3 from -1
level-controller.ts:523 [log] > [level-controller]: Attempt loading level index 3 with URL-id 0 https://test-streams.mux.dev/x36xhzz/url_0/193039199_mp4_h264_aac_hd_7.m3u8
base-stream-controller.ts:1265 [log] > [stream-controller]: STOPPED->IDLE
base-stream-controller.ts:1265 [log] > [subtitle-stream-controller]: IDLE->STOPPED
base-stream-controller.ts:1265 [log] > [subtitle-stream-controller]: STOPPED->IDLE
stream-controller.ts:604 [log] > [stream-controller]: Level 3 loaded [0,63], cc [0, 0] duration:634.584
buffer-controller.ts:628 [log] > [buffer-controller]: Updating Media Source duration to 634.584
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 0 cc: 0 of [0-63] level: 3, target: 0
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:66 [log] > demuxing in webworker
transmuxer-interface.ts:180 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 0 p: -1 level: 3 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 0
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 0 of level 3
​ [log] >
​ [log] > [mp4-remuxer]: ISGenerated flag reset
​ [log] > [mp4-remuxer]: initPTS & initDTS reset
​ [log] > manifest codec:mp4a.40.2, ADTS type:2, samplingIndex:4
​ [log] > parsed codec:mp4a.40.5, rate:44100, channels:2
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1207 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2/mp4a.40.2/mp4a.40.5]
stream-controller.ts:1218 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[avc1.64001f/avc1.64001f]
buffer-controller.ts:697 [log] > [buffer-controller]: creating sourceBuffer(audio/mp4;codecs=mp4a.40.2)
buffer-controller.ts:697 [log] > [buffer-controller]: creating sourceBuffer(video/mp4;codecs=avc1.64001f)
audio-stream-controller.ts:113 [log] > [audio-stream-controller]: InitPTS for cc: 0 found from main: 900909
​ [log] > [transmuxer.ts]: Flushed fragment 0 of level 3
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 0 of level 3 [0.023,10.008]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
level-controller.ts:250 [log] > [level-controller]: switching to level 4 from 3
level-controller.ts:523 [log] > [level-controller]: Attempt loading level index 4 with URL-id 0 https://test-streams.mux.dev/x36xhzz/url_8/193039199_mp4_h264_aac_fhd_7.m3u8
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->WAITING_LEVEL
stream-controller.ts:604 [log] > [stream-controller]: Level 4 loaded [0,63], cc [0, 0] duration:634.567
base-stream-controller.ts:1265 [log] > [stream-controller]: WAITING_LEVEL->IDLE
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 1 cc: 0 of [0-63] level: 4, target: 10.008
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:180 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 1 p: -1 level: 4 id: 1
        discontinuity: false
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 10
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 1 of level 4
​ [log] > [mp4-remuxer]: ISGenerated flag reset
​ [log] > [mp4-remuxer]: reset next timestamp
​ [log] > [mp4-remuxer]: ISGenerated flag reset
​ [log] > [mp4-remuxer]: initPTS & initDTS reset
​ [log] > manifest codec:mp4a.40.2, ADTS type:2, samplingIndex:4
​ [log] > parsed codec:mp4a.40.5, rate:44100, channels:2
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1207 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2/mp4a.40.2/mp4a.40.5]
stream-controller.ts:1218 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[avc1.640028/avc1.640028]
​ [log] > [transmuxer.ts]: Flushed fragment 1 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 1 of level 4 [0.023,19.992]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 2 cc: 0 of [0-63] level: 4, target: 19.992
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 2 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
​ [log] > [transmuxer.ts]: Flushed fragment 2 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 2 of level 4 [0.023,30.000]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 3 cc: 0 of [0-63] level: 4, target: 30
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 3 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
​ [log] > [transmuxer.ts]: Flushed fragment 3 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 3 of level 4 [0.023,40.008]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 4 cc: 0 of [0-63] level: 4, target: 40.008
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 4 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
​ [log] > [transmuxer.ts]: Flushed fragment 4 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 4 of level 4 [0.023,49.993]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 5 cc: 0 of [0-63] level: 4, target: 49.993
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 5 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
​ [log] > [transmuxer.ts]: Flushed fragment 5 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 5 of level 4 [0.023,60.000]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 6 cc: 0 of [0-63] level: 4, target: 60
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 6 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
​ [log] > [transmuxer.ts]: Flushed fragment 6 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 6 of level 4 [0.023,70.008]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 7 cc: 0 of [0-63] level: 4, target: 70.008
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 7 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
​ [log] > [transmuxer.ts]: Flushed fragment 7 of level 4
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 7 of level 4 [0.023,79.946]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
hls.ts:279 [log] > destroy
hls.ts:311 [log] > detachMedia
buffer-controller.ts:155 [log] > [buffer-controller]: media source detaching
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->STOPPED
base-stream-controller.ts:1265 [log] > [subtitle-stream-controller]: IDLE->STOPPED
main.js:337 Using Hls.js config: Object
logger.ts:74 [log] >
hls.ts:354 [log] > stopLoad
hls.ts:330 [log] > loadSource:https://os2man2.github.io/sample_hls_fmp4/master.m3u8
stream-controller.ts:540 [log] > [stream-controller]: Trigger BUFFER_RESET
hls.ts:302 [log] > attachMedia
level-controller.ts:171 [log] > [level-controller]: manifest loaded, 4 level(s) found, first bitrate: 211200
buffer-controller.ts:129 [log] > 2 bufferCodec event(s) expected
hls.ts:344 [log] > startLoad(-1)
level-controller.ts:250 [log] > [level-controller]: switching to level 0 from -1
audio-track-controller.ts:132 [log] > [audio-track-controller]: Updating audio tracks, 1 track(s) found in "l0" group-id
audio-track-controller.ts:188 [log] > [audio-track-controller]: Now switching to audio-track index 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: STOPPED->IDLE
level-controller.ts:523 [log] > [level-controller]: Attempt loading level index 0 with URL-id 0 https://os2man2.github.io/sample_hls_fmp4/0/output.m3u8
base-stream-controller.ts:1265 [log] > [stream-controller]: STOPPED->IDLE
audio-track-controller.ts:248 [log] > [audio-track-controller]: loading audio-track playlist for id: 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->STOPPED
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: STOPPED->WAITING_TRACK
base-stream-controller.ts:1265 [log] > [subtitle-stream-controller]: STOPPED->IDLE
buffer-controller.ts:736 [log] > [buffer-controller]: Media source opened
stream-controller.ts:604 [log] > [stream-controller]: Level 0 loaded [0,7], cc [0, 0] duration:15
buffer-controller.ts:628 [log] > [buffer-controller]: Updating Media Source duration to 15.000
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment initSegment cc: 0 level: 0, target: 0
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
audio-track-controller.ts:85 [log] > [audio-track-controller]: audioTrack 0 loaded [0-7]
audio-stream-controller.ts:440 [log] > [audio-stream-controller]: Track 0 loaded [0,7],duration:14.512471000000001
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: WAITING_TRACK->IDLE
base-stream-controller.ts:555 [log] > [audio-stream-controller]: Loading fragment initSegment cc: 0 track: 0, target: 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->IDLE
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: initSegment of level 0 
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 0 cc: 0 of [0-7] level: 0, target: 0
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: FRAG_LOADING->IDLE
base-stream-controller.ts:466 [log] > [audio-stream-controller]: Buffered audio sn: initSegment of track 0 
base-stream-controller.ts:555 [log] > [audio-stream-controller]: Loading fragment 0 cc: 0 of [0-7] track: 0, target: 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:66 [log] > demuxing in webworker
transmuxer-interface.ts:180 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 0 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 0
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 0 of level 0
transmuxer-interface.ts:66 [log] > demuxing in webworker
audio-stream-controller.ts:544 [log] > Unknown video PTS for cc 0, waiting for video PTS before demuxing audio frag 0 of [0 ,7],track 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: FRAG_LOADING->WAITING_INIT_PTS
base-stream-controller.ts:329 [log] > [audio-stream-controller]: Loaded fragment 0 of level 0
hls.ts:279 [log] > destroy
hls.ts:311 [log] > detachMedia
buffer-controller.ts:155 [log] > [buffer-controller]: media source detaching
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->STOPPED
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: WAITING_INIT_PTS->STOPPED
base-stream-controller.ts:1265 [log] > [subtitle-stream-controller]: IDLE->STOPPED
main.js:337 Using Hls.js config: Object
logger.ts:74 [log] >
hls.ts:354 [log] > stopLoad
hls.ts:330 [log] > loadSource:https://os2man2.github.io/sample_hls_fmp4/master.m3u8
stream-controller.ts:540 [log] > [stream-controller]: Trigger BUFFER_RESET
hls.ts:302 [log] > attachMedia
level-controller.ts:171 [log] > [level-controller]: manifest loaded, 4 level(s) found, first bitrate: 211200
buffer-controller.ts:129 [log] > 2 bufferCodec event(s) expected
hls.ts:344 [log] > startLoad(-1)
level-controller.ts:250 [log] > [level-controller]: switching to level 0 from -1
audio-track-controller.ts:132 [log] > [audio-track-controller]: Updating audio tracks, 1 track(s) found in "l0" group-id
audio-track-controller.ts:188 [log] > [audio-track-controller]: Now switching to audio-track index 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: STOPPED->IDLE
level-controller.ts:523 [log] > [level-controller]: Attempt loading level index 0 with URL-id 0 https://os2man2.github.io/sample_hls_fmp4/0/output.m3u8
base-stream-controller.ts:1265 [log] > [stream-controller]: STOPPED->IDLE
audio-track-controller.ts:248 [log] > [audio-track-controller]: loading audio-track playlist for id: 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->STOPPED
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: STOPPED->WAITING_TRACK
base-stream-controller.ts:1265 [log] > [subtitle-stream-controller]: STOPPED->IDLE
buffer-controller.ts:736 [log] > [buffer-controller]: Media source opened
stream-controller.ts:604 [log] > [stream-controller]: Level 0 loaded [0,7], cc [0, 0] duration:15
buffer-controller.ts:628 [log] > [buffer-controller]: Updating Media Source duration to 15.000
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment initSegment cc: 0 level: 0, target: 0
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
audio-track-controller.ts:85 [log] > [audio-track-controller]: audioTrack 0 loaded [0-7]
audio-stream-controller.ts:440 [log] > [audio-stream-controller]: Track 0 loaded [0,7],duration:14.512471000000001
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: WAITING_TRACK->IDLE
base-stream-controller.ts:555 [log] > [audio-stream-controller]: Loading fragment initSegment cc: 0 track: 0, target: 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->IDLE
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: initSegment of level 0 
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 0 cc: 0 of [0-7] level: 0, target: 0
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: FRAG_LOADING->IDLE
base-stream-controller.ts:466 [log] > [audio-stream-controller]: Buffered audio sn: initSegment of track 0 
base-stream-controller.ts:555 [log] > [audio-stream-controller]: Loading fragment 0 cc: 0 of [0-7] track: 0, target: 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:66 [log] > demuxing in webworker
transmuxer-interface.ts:180 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 0 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 0
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 0 of level 0
transmuxer-interface.ts:66 [log] > demuxing in webworker
audio-stream-controller.ts:544 [log] > Unknown video PTS for cc 0, waiting for video PTS before demuxing audio frag 0 of [0 ,7],track 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: FRAG_LOADING->WAITING_INIT_PTS
base-stream-controller.ts:329 [log] > [audio-stream-controller]: Loaded fragment 0 of level 0
eacf9fa6-d196-4dc4-9d02-d95e2612c10d:603 [log] >
2f88ac54-3225-4064-9822-fa938787ec05:603 [log] >
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
eacf9fa6-d196-4dc4-9d02-d95e2612c10d:935 [log] > [transmuxer.ts]: Flushed fragment 0 of level 0
stream-controller.ts:1218 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[avc1.641028/avc1.641028]
audio-stream-controller.ts:113 [log] > [audio-stream-controller]: InitPTS for cc: 0 found from main: 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: WAITING_INIT_PTS->FRAG_LOADING
transmuxer-interface.ts:180 [log] > [transmuxer-interface, audio]: Starting new transmux session for sn: 0 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: false
        timeOffset: 0
2f88ac54-3225-4064-9822-fa938787ec05:935 [log] > [transmuxer.ts]: Flushed fragment 0 of level 0
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: FRAG_LOADING->PARSING
audio-stream-controller.ts:829 [log] > [audio-stream-controller]: Switching audio track : flushing all audio
audio-stream-controller.ts:777 [log] > [audio-stream-controller]: Init audio buffer, container:audio/mp4, codecs[parsed]=[mp4a.40.2]
buffer-controller.ts:697 [log] > [buffer-controller]: creating sourceBuffer(video/mp4;codecs=avc1.641028)
buffer-controller.ts:697 [log] > [buffer-controller]: creating sourceBuffer(audio/mp4;codecs=mp4a.40.2)
stream-controller.ts:819 [log] > [stream-controller]: Alternate track found, use video.buffered to schedule main fragment loading
buffer-controller.ts:810 [log] > [buffer-controller]: Removing [0,15] from the audio SourceBuffer
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 0 of level 0 [0.000,2.000]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
level-controller.ts:250 [log] > [level-controller]: switching to level 3 from 0
audio-track-controller.ts:132 [log] > [audio-track-controller]: Updating audio tracks, 1 track(s) found in "l3" group-id
audio-track-controller.ts:188 [log] > [audio-track-controller]: Now switching to audio-track index 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: PARSED->IDLE
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->WAITING_TRACK
audio-track-controller.ts:248 [log] > [audio-track-controller]: loading audio-track playlist for id: 0
level-controller.ts:523 [log] > [level-controller]: Attempt loading level index 3 with URL-id 0 https://os2man2.github.io/sample_hls_fmp4/6/output.m3u8
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->WAITING_LEVEL
audio-stream-controller.ts:591 [warn] > [audio-stream-controller]: Fragment 0 of level 0 finished buffering, but was aborted. state: WAITING_TRACK, audioSwitch: true
onFragBuffered @ audio-stream-controller.ts:591
stream-controller.ts:604 [log] > [stream-controller]: Level 3 loaded [0,7], cc [0, 0] duration:15
base-stream-controller.ts:1265 [log] > [stream-controller]: WAITING_LEVEL->IDLE
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment initSegment cc: 0 level: 3, target: 0
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
audio-track-controller.ts:85 [log] > [audio-track-controller]: audioTrack 0 loaded [0-7]
audio-stream-controller.ts:440 [log] > [audio-stream-controller]: Track 0 loaded [0,7],duration:14.512471000000001
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: WAITING_TRACK->IDLE
base-stream-controller.ts:555 [log] > [audio-stream-controller]: Loading fragment initSegment cc: 0 track: 0, target: 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->IDLE
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: initSegment of level 3 [0.000,2.000]
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 1 cc: 0 of [0-7] level: 3, target: 2
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
/favicon.ico:1 Failed to load resource: the server responded with a status of 404 ()
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: FRAG_LOADING->IDLE
base-stream-controller.ts:466 [log] > [audio-stream-controller]: Buffered audio sn: initSegment of track 0 [0.000,2.020]
base-stream-controller.ts:555 [log] > [audio-stream-controller]: Loading fragment 0 cc: 0 of [0-7] track: 0, target: 0.158
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:180 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 1 p: -1 level: 3 id: 1
        discontinuity: false
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 2
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 1 of level 3
eacf9fa6-d196-4dc4-9d02-d95e2612c10d:935 [log] > [transmuxer.ts]: Flushed fragment 1 of level 3
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1218 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[avc1.641028/avc1.641028]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:329 [log] > [audio-stream-controller]: Loaded fragment 0 of level 0
2f88ac54-3225-4064-9822-fa938787ec05:935 [log] > [transmuxer.ts]: Flushed fragment 0 of level 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: FRAG_LOADING->PARSING
audio-stream-controller.ts:829 [log] > [audio-stream-controller]: Switching audio track : flushing all audio
buffer-controller.ts:810 [log] > [buffer-controller]: Removing [0,15] from the audio SourceBuffer
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 1 of level 3 [0.000,4.000]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
level-controller.ts:250 [log] > [level-controller]: switching to level 2 from 3
audio-track-controller.ts:132 [log] > [audio-track-controller]: Updating audio tracks, 1 track(s) found in "l2" group-id
audio-track-controller.ts:188 [log] > [audio-track-controller]: Now switching to audio-track index 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: PARSED->IDLE
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->WAITING_TRACK
audio-track-controller.ts:248 [log] > [audio-track-controller]: loading audio-track playlist for id: 0
level-controller.ts:523 [log] > [level-controller]: Attempt loading level index 2 with URL-id 0 https://os2man2.github.io/sample_hls_fmp4/4/output.m3u8
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->WAITING_LEVEL
audio-stream-controller.ts:591 [warn] > [audio-stream-controller]: Fragment 0 of level 0 finished buffering, but was aborted. state: WAITING_TRACK, audioSwitch: true
onFragBuffered @ audio-stream-controller.ts:591
audio-track-controller.ts:85 [log] > [audio-track-controller]: audioTrack 0 loaded [0-7]
audio-stream-controller.ts:440 [log] > [audio-stream-controller]: Track 0 loaded [0,7],duration:14.512471000000001
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: WAITING_TRACK->IDLE
base-stream-controller.ts:555 [log] > [audio-stream-controller]: Loading fragment initSegment cc: 0 track: 0, target: 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
stream-controller.ts:604 [log] > [stream-controller]: Level 2 loaded [0,7], cc [0, 0] duration:15
base-stream-controller.ts:1265 [log] > [stream-controller]: WAITING_LEVEL->IDLE
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment initSegment cc: 0 level: 2, target: 0
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: FRAG_LOADING->IDLE
base-stream-controller.ts:466 [log] > [audio-stream-controller]: Buffered audio sn: initSegment of track 0 [0.000,2.020]
base-stream-controller.ts:555 [log] > [audio-stream-controller]: Loading fragment 0 cc: 0 of [0-7] track: 0, target: 0.881
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->IDLE
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: initSegment of level 2 [0.000,4.000]
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 2 cc: 0 of [0-7] level: 2, target: 4
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:329 [log] > [audio-stream-controller]: Loaded fragment 0 of level 0
2f88ac54-3225-4064-9822-fa938787ec05:935 [log] > [transmuxer.ts]: Flushed fragment 0 of level 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: FRAG_LOADING->PARSING
audio-stream-controller.ts:829 [log] > [audio-stream-controller]: Switching audio track : flushing all audio
buffer-controller.ts:810 [log] > [buffer-controller]: Removing [0,15] from the audio SourceBuffer
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [audio-stream-controller]: Buffered audio sn: 0 of track 0 [0.000,2.020]
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: PARSED->IDLE
transmuxer-interface.ts:180 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 2 p: -1 level: 2 id: 1
        discontinuity: false
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 4
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 2 of level 2
eacf9fa6-d196-4dc4-9d02-d95e2612c10d:935 [log] > [transmuxer.ts]: Flushed fragment 2 of level 2
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1218 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[avc1.641028/avc1.641028]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 2 of level 2 [0.000,6.000]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
level-controller.ts:250 [log] > [level-controller]: switching to level 3 from 2
audio-track-controller.ts:132 [log] > [audio-track-controller]: Updating audio tracks, 1 track(s) found in "l3" group-id
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 3 cc: 0 of [0-7] level: 3, target: 6
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:180 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 3 p: -1 level: 3 id: 1
        discontinuity: false
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 6
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 3 of level 3
eacf9fa6-d196-4dc4-9d02-d95e2612c10d:935 [log] > [transmuxer.ts]: Flushed fragment 3 of level 3
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1218 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[avc1.641028/avc1.641028]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 3 of level 3 [0.000,8.000]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 4 cc: 0 of [0-7] level: 3, target: 8
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 4 of level 3
eacf9fa6-d196-4dc4-9d02-d95e2612c10d:935 [log] > [transmuxer.ts]: Flushed fragment 4 of level 3
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 4 of level 3 [0.000,10.000]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
level-controller.ts:250 [log] > [level-controller]: switching to level 2 from 3
audio-track-controller.ts:132 [log] > [audio-track-controller]: Updating audio tracks, 1 track(s) found in "l2" group-id
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 5 cc: 0 of [0-7] level: 2, target: 10
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:180 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 5 p: -1 level: 2 id: 1
        discontinuity: false
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 10
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 5 of level 2
eacf9fa6-d196-4dc4-9d02-d95e2612c10d:935 [log] > [transmuxer.ts]: Flushed fragment 5 of level 2
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1218 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[avc1.641028/avc1.641028]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 5 of level 2 [0.000,12.000]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
level-controller.ts:250 [log] > [level-controller]: switching to level 0 from 2
audio-track-controller.ts:132 [log] > [audio-track-controller]: Updating audio tracks, 1 track(s) found in "l0" group-id
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 6 cc: 0 of [0-7] level: 0, target: 12
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
gap-controller.ts:208 [warn] > Playback stalling at @1.945726 due to low buffer (buffer=0.07440899999999973)
_reportStall @ gap-controller.ts:208
latency-controller.ts:188 [warn] > [playback-rate-controller]: Stall detected, adjusting target latency
onError @ latency-controller.ts:188
main.js:721 Error event: Object
(anonymous) @ main.js:721
base-stream-controller.ts:555 [log] > [audio-stream-controller]: Loading fragment 1 cc: 0 of [0-7] track: 0, target: 2.02
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:180 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 6 p: -1 level: 0 id: 1
        discontinuity: false
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 12
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 6 of level 0
eacf9fa6-d196-4dc4-9d02-d95e2612c10d:935 [log] > [transmuxer.ts]: Flushed fragment 6 of level 0
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1218 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[avc1.641028/avc1.641028]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:329 [log] > [audio-stream-controller]: Loaded fragment 1 of level 0
2f88ac54-3225-4064-9822-fa938787ec05:935 [log] > [transmuxer.ts]: Flushed fragment 1 of level 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: FRAG_LOADING->PARSING
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 6 of level 0 [0.000,14.000]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
level-controller.ts:250 [log] > [level-controller]: switching to level 3 from 0
audio-track-controller.ts:132 [log] > [audio-track-controller]: Updating audio tracks, 1 track(s) found in "l3" group-id
base-stream-controller.ts:555 [log] > [stream-controller]: Loading fragment 7 cc: 0 of [0-7] level: 3, target: 14
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:466 [log] > [audio-stream-controller]: Buffered audio sn: 1 of track 0 [0.000,4.017]
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: PARSED->IDLE
base-stream-controller.ts:962 [log] > [audio-stream-controller]: SN 1 just loaded, load next one: 2
base-stream-controller.ts:555 [log] > [audio-stream-controller]: Loading fragment 2 cc: 0 of [0-7] track: 0, target: 6.037
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
gap-controller.ts:60 [warn] > playback not stuck anymore @2.089026, after 737ms
poll @ gap-controller.ts:60
transmuxer-interface.ts:180 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 7 p: -1 level: 3 id: 1
        discontinuity: false
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 14
base-stream-controller.ts:329 [log] > [stream-controller]: Loaded fragment 7 of level 3
eacf9fa6-d196-4dc4-9d02-d95e2612c10d:935 [log] > [transmuxer.ts]: Flushed fragment 7 of level 3
base-stream-controller.ts:1265 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1218 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[avc1.641028/avc1.641028]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:329 [log] > [audio-stream-controller]: Loaded fragment 2 of level 0
2f88ac54-3225-4064-9822-fa938787ec05:935 [log] > [transmuxer.ts]: Flushed fragment 2 of level 0
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: FRAG_LOADING->PARSING
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: PARSING->PARSED
base-stream-controller.ts:466 [log] > [stream-controller]: Buffered main sn: 7 of level 3 [0.000,15.000]
base-stream-controller.ts:1265 [log] > [stream-controller]: PARSED->IDLE
buffer-controller.ts:501 [log] > [buffer-controller]: video sourceBuffer now EOS
base-stream-controller.ts:1265 [log] > [stream-controller]: IDLE->ENDED
base-stream-controller.ts:466 [log] > [audio-stream-controller]: Buffered audio sn: 2 of track 0 [0.000,6.014]
base-stream-controller.ts:1265 [log] > [audio-stream-controller]: PARSED->IDLE
gap-controller.ts:208 [warn] > Playback stalling at @5.943839 due to low buffer (buffer=0.07012800000000041)
_reportStall @ gap-controller.ts:208
latency-controller.ts:188 [warn] > [playback-rate-controller]: Stall detected, adjusting target latency
onError @ latency-controller.ts:188
main.js:721 Error event: Object
(anonymous) @ main.js:721
```

--- a/api-extractor/report/hls.js.api.md
+++ b/api-extractor/report/hls.js.api.md
@@ -82,6 +82,8 @@ export interface AudioTrackSwitchedData {
 // @public (undocumented)
 export interface AudioTrackSwitchingData {
     // (undocumented)
+    groupId: string;
+    // (undocumented)
     id: number;
     // (undocumented)
     name: string;
@@ -2007,6 +2009,8 @@ export interface SubtitleTracksUpdatedData {
 //
 // @public (undocumented)
 export interface SubtitleTrackSwitchData {
+    // (undocumented)
+    groupId?: string;
     // (undocumented)
     id: number;
     // (undocumented)

--- a/src/controller/audio-stream-controller.ts
+++ b/src/controller/audio-stream-controller.ts
@@ -7,7 +7,6 @@ import { FragmentState } from './fragment-tracker';
 import { Level } from '../types/level';
 import { PlaylistLevelType } from '../types/loader';
 import { Fragment, ElementaryStreamTypes, Part } from '../loader/fragment';
-import FragmentLoader from '../loader/fragment-loader';
 import ChunkCache from '../demux/chunk-cache';
 import TransmuxerInterface from '../demux/transmuxer-interface';
 import type { TransmuxerResult } from '../types/transmuxer';
@@ -364,6 +363,7 @@ class AudioStreamController
     event: Events.AUDIO_TRACKS_UPDATED,
     { audioTracks }: AudioTracksUpdatedData
   ) {
+    this.resetTransmuxer();
     this.levels = audioTracks.map((mediaPlaylist) => new Level(mediaPlaylist));
   }
 
@@ -374,7 +374,7 @@ class AudioStreamController
     // if any URL found on new audio track, it is an alternate audio track
     const altAudio = !!data.url;
     this.trackId = data.id;
-    const { fragCurrent, transmuxer } = this;
+    const { fragCurrent } = this;
 
     if (fragCurrent?.loader) {
       fragCurrent.loader.abort();
@@ -383,10 +383,7 @@ class AudioStreamController
     this.clearWaitingFragment();
     // destroy useless transmuxer when switching audio to main
     if (!altAudio) {
-      if (transmuxer) {
-        transmuxer.destroy();
-        this.transmuxer = null;
-      }
+      this.resetTransmuxer();
     } else {
       // switching to audio track, start timer if not already started
       this.setInterval(TICK_INTERVAL);

--- a/src/controller/audio-track-controller.ts
+++ b/src/controller/audio-track-controller.ts
@@ -188,11 +188,17 @@ class AudioTrackController extends BasePlaylistController {
 
     const lastTrack = tracks[this.trackId];
     this.log(`Now switching to audio-track index ${newId}`);
-    const { id, name, type, url } = track;
+    const { id, groupId = '', name, type, url } = track;
     this.trackId = newId;
     this.trackName = name;
     this.selectDefaultTrack = false;
-    this.hls.trigger(Events.AUDIO_TRACK_SWITCHING, { id, name, type, url });
+    this.hls.trigger(Events.AUDIO_TRACK_SWITCHING, {
+      id,
+      groupId,
+      name,
+      type,
+      url,
+    });
     const hlsUrlParameters = this.switchParams(track.url, lastTrack?.details);
     this.loadPlaylist(hlsUrlParameters);
   }

--- a/src/controller/base-stream-controller.ts
+++ b/src/controller/base-stream-controller.ts
@@ -121,10 +121,7 @@ export default class BaseStreamController
     if (frag) {
       this.fragmentTracker.removeFragment(frag);
     }
-    if (this.transmuxer) {
-      this.transmuxer.destroy();
-      this.transmuxer = null;
-    }
+    this.resetTransmuxer();
     this.fragCurrent = null;
     this.fragPrevious = null;
     this.clearInterval();
@@ -1228,10 +1225,7 @@ export default class BaseStreamController
             this.warn(
               `Could not parse fragment ${frag.sn} ${type} duration reliably (${parsedDuration}) resetting transmuxer to fallback to playlist timing`
             );
-            if (this.transmuxer) {
-              this.transmuxer.destroy();
-              this.transmuxer = null;
-            }
+            this.resetTransmuxer();
             return result || false;
           }
           const drift = partial
@@ -1266,6 +1260,13 @@ export default class BaseStreamController
       this.fragCurrent = null;
       this.fragPrevious = null;
       this.state = State.IDLE;
+    }
+  }
+
+  protected resetTransmuxer() {
+    if (this.transmuxer) {
+      this.transmuxer.destroy();
+      this.transmuxer = null;
     }
   }
 

--- a/src/controller/stream-controller.ts
+++ b/src/controller/stream-controller.ts
@@ -715,13 +715,6 @@ export default class StreamController
     );
   }
 
-  private resetTransmuxer() {
-    if (this.transmuxer) {
-      this.transmuxer.destroy();
-      this.transmuxer = null;
-    }
-  }
-
   private onAudioTrackSwitching(
     event: Events.AUDIO_TRACK_SWITCHING,
     data: AudioTrackSwitchingData

--- a/src/controller/subtitle-stream-controller.ts
+++ b/src/controller/subtitle-stream-controller.ts
@@ -5,7 +5,6 @@ import { findFragmentByPDT, findFragmentByPTS } from './fragment-finders';
 import type { FragmentTracker } from './fragment-tracker';
 import { FragmentState } from './fragment-tracker';
 import BaseStreamController, { State } from './base-stream-controller';
-import FragmentLoader from '../loader/fragment-loader';
 import { PlaylistLevelType } from '../types/loader';
 import { Level } from '../types/level';
 import type { NetworkComponentAPI } from '../types/component-api';

--- a/src/controller/subtitle-track-controller.ts
+++ b/src/controller/subtitle-track-controller.ts
@@ -354,8 +354,14 @@ class SubtitleTrackController extends BasePlaylistController {
     this.log(`Switching to subtitle track ${newId}`);
     this.trackId = newId;
     if (track) {
-      const { id, name, type, url } = track;
-      this.hls.trigger(Events.SUBTITLE_TRACK_SWITCH, { id, name, type, url });
+      const { id, groupId = '', name, type, url } = track;
+      this.hls.trigger(Events.SUBTITLE_TRACK_SWITCH, {
+        id,
+        groupId,
+        name,
+        type,
+        url,
+      });
       const hlsUrlParameters = this.switchParams(track.url, lastTrack?.details);
       this.loadPlaylist(hlsUrlParameters);
     } else {

--- a/src/types/events.ts
+++ b/src/types/events.ts
@@ -155,6 +155,7 @@ export interface LevelPTSUpdatedData {
 export interface AudioTrackSwitchingData {
   id: number;
   name: string;
+  groupId: string;
   type: MediaPlaylistType | 'main';
   url: string;
 }
@@ -176,6 +177,7 @@ export interface SubtitleTracksUpdatedData {
 export interface SubtitleTrackSwitchData {
   id: number;
   name?: string;
+  groupId?: string;
   type?: MediaPlaylistType | 'main';
   url?: string;
 }

--- a/tests/unit/controller/subtitle-track-controller.js
+++ b/tests/unit/controller/subtitle-track-controller.js
@@ -129,6 +129,7 @@ describe('SubtitleTrackController', function () {
         'hlsSubtitleTrackSwitch',
         {
           id: 1,
+          groupId: 'default-text-group',
           name: 'English',
           type: 'SUBTITLES',
           url: 'bar',
@@ -164,6 +165,7 @@ describe('SubtitleTrackController', function () {
         'hlsSubtitleTrackSwitch',
         {
           id: 0,
+          groupId: 'default-text-group',
           name: 'English',
           type: 'SUBTITLES',
           url: 'baz',

